src/main/scala/com/imaginea/activegrid/core/models/AnsiblePlayBookRunner.scala

```diff
@@ -1,24 +1,43 @@
 package com.imaginea.activegrid.core.models
+
 import com.typesafe.scalalogging.Logger
 import org.slf4j.LoggerFactory
-import scala.util.Try
+
+import scala.util.{Failure, Success}
+
 /**
   * Created by sivag on 17/1/17.
   */
 class AnsiblePlayBookRunner(workflowContext: WorkflowContext) {
+
   val logger = Logger(LoggerFactory.getLogger(AnsiblePlayBookRunner.this.getClass.getName))
-  def executePlayBook() : Try[Boolean] = {
-    //todo implementation.
-    Try(false) //dummy response.
-  }
-  def execute() : Try[Boolean] = {
-     executePlayBook() recover {
-       case e : Exception =>
-         val workflowId = WorkflowContext.get().workflow.id
-       logger.error(s"An unexpected error, Execution of workflow ($workflowId) aborted., Removig workflow from running list",e)
-       CurrentRunningWorkflows.remove(workflowId.getOrElse(0L));
-     }
-    Try(true);
+
+  def execute(): Unit = {
+    val workflow = workflowContext.workflow
+    val execution = workflow.execution.get
```

Comment:
.get doesn't fail in the above case. because in the below call hierarchy, control will forward only if execution exists.

service.execute(){ if(**execution eixsts)** { -->  **ansibleservice.execute()** --->  **playbookrunner.execute()** //current method }